### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by chepner** [koalaman/shellcheck#1722]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

a=

if [ -z $a ]; then echo pass; fi
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

The lack of a warning here (even though `[ -z ]` and `[ -z "" ]` have the same results, though for different reasons) feels inconsistent. I acknowledge that a warning *is* raised when the value of `a` cannot be statically determined.

**Comment by koalaman** [comment]
Wow, great catch. ShellCheck just checked whether the variable might contain spaces, which an empty variable obviously can't. This check will now additionally track whether the variable may be empty, so that this warning will still trigger. Thanks!
